lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -299,7 +320,34 @@ def extract_token_from_string_and_maybe_add_customer_id(post, payment_method)
           post[:customer] = customer_id
         end
 
-        post[:payment_method] = payment_method
+        if payment_method.include?('tok_')
+          add_payment_method_card_data_tok(post, payment_method)
+        else
+          post[:payment_method] = payment_method
+        end
+      end
+
+      def tokenize_apple_google(payment, options = {})
+        tokenization_method = options[:tokenization_method] || 'android_pay'
+        tokenization_method = 'apple_pay' if payment.source.to_sym == :apple_pay
+        post = {
+          card: {
+            number: payment.number,
+            exp_month: payment.month,
+            exp_year: payment.year,
+            tokenization_method: tokenization_method,
+            eci: payment.eci,
+            cryptogram: payment.payment_cryptogram
+          }
+        }
+        token_response = api_request(:post, 'tokens', post, {})
+        success = token_response['error'].nil?
+
+        if success && token_response['id']
+          Response.new(success, nil, token: token_response)
+        else
```

Comment:
❓  Question:
 Is possible to have a success response that doesn't have an 'id' key ?, in that case we are going to return an error with out a message